File: ios/Classes/Clearsale.swift
//
// Created by Development on 16/06/21.
//

import CSBehavior

public class Clearsale {
    private var mInstance: CSBehavior;

    init(appkey: String?){
        mInstance = CSBehavior.getInstance(appkey);
    }

    public func generateSessionId() -> String? {
        return mInstance.generateSessionId();
    }

    public func collectDeviceInformation(sessionId: String){
        mInstance.collectDeviceInformation(sessionId);
    }
}

File: android/src/main/kotlin/com/br/dblinks/clearsale_fingerprint/Clearsale.kt
package com.br.dblinks.clearsale_fingerprint

import android.content.Context
import sale.clear.behavior.android.Behavior

class Clearsale(context: Context?, appKey: String?) {
    private var mInstance: Behavior = Behavior.getInstance(context, appKey);

    fun start() {
        this.mInstance.start();
    }

    fun stop() {
        this.mInstance.stop();
    }

    fun generateSessionID(): String? {
        return this.mInstance.generateSessionID();
    }

    fun collectDeviceInformation(sessionId: String) {
        mInstance.collectDeviceInformation(sessionId);
    }

    fun getInstance(): Behavior {
        return this.mInstance;
    }
}